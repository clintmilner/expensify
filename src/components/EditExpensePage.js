import React from 'react';

const EditExpensePage = (props) => {
    console.log('props', props);
    return (
        <div>
            Editing Expense #{props.match.params.id}
        </div>
    );
};

export default EditExpensePage;