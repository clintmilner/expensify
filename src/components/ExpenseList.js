import React from 'react';
import { connect } from 'react-redux';

const ExpenseList = (props) => (
    <div>
        Expense List is Here {props.expenses.length} - {props.filters.text}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    };
};

export default connect( mapStateToProps )( ExpenseList );
