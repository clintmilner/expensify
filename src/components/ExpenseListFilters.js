import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate} from '../actions/filters';

const ExpenseListFilters = (props) => (
    <div>
        <input type='text' value={props.filters.text} onChange={ (e) => {
            props.dispatch( setTextFilter(e.target.value) );
        }}  />
        <select value={props.filters.sortBy}
                onChange={(e) => {
                    const type = e.target.value;
                    switch (type) {
                        case 'date':
                            props.dispatch( sortByDate( type ) );
                            break;
                        case 'amount':
                            props.dispatch( sortByAmount( type ) );
                            break;
                        default:
                    }
                }
            }>
            <option value='date'>Date</option>
            <option value='amount'>Amount</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return (
        {
            filters: state.filters
        }
    )
};
export default connect(mapStateToProps)(ExpenseListFilters);