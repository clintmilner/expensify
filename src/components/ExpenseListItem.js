import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = (props) => {
    const {key, id, description, amount, createdAt} = props.expense;
    return (
        <div key={key}>
            <h3>{description}</h3>
            <p>{amount} - {createdAt}</p>
            <button onClick={() => {
                props.dispatch(removeExpense( {id} ));
            }}>Remove Expense</button>
        </div>
    );
};

export default connect()(ExpenseListItem); // we don't need state, so connect() can be empty!