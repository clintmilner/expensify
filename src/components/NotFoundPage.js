import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        Page Not Found 404 - <Link to='/'>Go Home, You're Drunk</Link>
    </div>
);

export default NotFoundPage;