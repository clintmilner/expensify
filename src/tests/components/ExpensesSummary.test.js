import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render correctly with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={311}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={311816420}/>);
    expect(wrapper).toMatchSnapshot();
});