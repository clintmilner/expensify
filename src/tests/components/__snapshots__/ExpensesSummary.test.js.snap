// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $3,118,164.20
  </h1>
</div>
`;

exports[`should render correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $3.11
  </h1>
</div>
`;
